src/pmdas/bpf/modules/execsnoop.c

```diff
@@ -0,0 +1,395 @@
+        }
+    };
+    /* bpf.execsnoop.pid */
+    metrics[PID] = (struct pmdaMetric)
+    { /* m_user */ NULL,
+        { /* m_desc */
+            PMDA_PMID(cluster_id, 1), PM_TYPE_U32, indom_id_mapping[EXECSNOOP_INDOM],
+            PM_SEM_INSTANT, PMDA_PMUNITS(0, 0, 0, 0, 0, 0)
+        }
+    };
+    /* bpf.execsnoop.ppid */
+    metrics[PPID] = (struct pmdaMetric)
+    { /* m_user */ NULL,
+        { /* m_desc */
+            PMDA_PMID(cluster_id, 2), PM_TYPE_U32, indom_id_mapping[EXECSNOOP_INDOM],
+            PM_SEM_INSTANT, PMDA_PMUNITS(0, 0, 0, 0, 0, 0)
+        }
+    };
+    /* bpf.execsnoop.ret */
+    metrics[RET] = (struct pmdaMetric)
+    { /* m_user */ NULL,
+        { /* m_desc */
+            PMDA_PMID(cluster_id, 3), PM_TYPE_U32, indom_id_mapping[EXECSNOOP_INDOM],
+            PM_SEM_INSTANT, PMDA_PMUNITS(0, 0, 0, 0, 0, 0)
+        }
+    };
+    /* bpf.execsnoop.args */
+    metrics[ARGS] = (struct pmdaMetric)
+    { /* m_user */ NULL,
+        { /* m_desc */
+            PMDA_PMID(cluster_id, 4), PM_TYPE_STRING, indom_id_mapping[EXECSNOOP_INDOM],
+            PM_SEM_INSTANT, PMDA_PMUNITS(0, 0, 0, 0, 0, 0)
+        }
+    };
+    /* bpf.execsnoop.uid */
+    metrics[UID] = (struct pmdaMetric)
+    { /* m_user */ NULL,
+        { /* m_desc */
+            PMDA_PMID(cluster_id, 5), PM_TYPE_U32, indom_id_mapping[EXECSNOOP_INDOM],
+            PM_SEM_INSTANT, PMDA_PMUNITS(0, 0, 0, 0, 0, 0)
+        }
+    };
+
+    indoms[0] = (struct pmdaIndom)
+    {
+        indom_id_mapping[EXECSNOOP_INDOM],
+        env.process_count,
+        execsnoop_instances,
+    };
+}
+
+static void handle_args(const struct event *e)
+{
+    int i, args_counter = 0;
+
+    for (i = 0; i < e->args_size && args_counter < e->args_count; i++) {
+        char c = e->args[i];
+        if (c == '\0') {
+            args_counter++;
+            arg_val[i] = ' ';
+        } else {
+            arg_val[i] = c;
+        }
+    }
+    if (e->args_count == env.max_args + 1) {
+        arg_val[e->args_size-1] = '\0';
+    }
+    arg_val[e->args_size-1] = '\0';
+}
+
+static void handle_event(void *ctx, int cpu, void *data, __u32 data_sz)
+{
+    event = data;
+    handle_args(event);
+
+    struct tailq_entry *elm = allocElm();
+
+    elm->event.pid = event->pid;
+    elm->event.ppid = event->ppid;
+    elm->event.uid = event->uid;
+    elm->event.retval = event->retval;
+    elm->event.args_count = event->args_count;
+    elm->event.args_size = event->args_size;
+    strcpy(elm->event.comm, event->comm);
+    strcpy(elm->event.args, arg_val);
+
+    /* TODO: use pcre lib */
+    if (env.name && strstr(elm->event.comm, env.name) == NULL)
+        return;
+
+    /* TODO: use pcre lib */
+    if (env.line && strstr(elm->event.comm, env.line) == NULL)
+        return;
+
+    push(elm);
+}
+
+static void handle_lost_events(void *ctx, int cpu, __u64 lost_cnt)
+{
+    pmNotifyErr(LOG_ERR, "Lost %llu events on CPU #%d!", lost_cnt, cpu);
```

Comment:
Nice Note, Just added one commit for this, I added `bpf.execsnoop.lost`  in a separate InDom has one instance represent the count of lost events.
Expected output: ( because I don't know how I could make my system busy enough to drop some events!)
```
$ pminfo -f bpf.execsnoop.lost

bpf.execsnoop.lost
    inst [0 or "0"] value 0
```
I know. It's not that much perfect, I think It will be better if I did something like that:
multiple instance each one represents one CPU (hence external instance name)
```
$ pminfo -f bpf.execsnoop.lost

bpf.execsnoop.lost
    inst [0 or "CPU 0"] value 10
    inst [0 or "CPU 10"] value 4
    inst [0 or "CPU 100"] value 20
```
but since I didn't test that even with iovisor/execsnoop, I couldn't handle that,
It seems not a big deal but It will need surgery ^^"
Maybe I could handle this later after merging this PR...

Thanks : )